server/src/test/java/org/opensearch/cluster/service/MasterTaskThrottlerTests.java

```diff
@@ -0,0 +1,228 @@
+            buildNewFakeTransportAddress(),
+            Collections.emptyMap(),
+            Collections.singleton(DiscoveryNodeRole.CLUSTER_MANAGER_ROLE),
+            Version.V_3_0_0
+        );
+        DiscoveryNode dataNode = new DiscoveryNode(
+            "local_data_node",
+            buildNewFakeTransportAddress(),
+            Collections.emptyMap(),
+            Collections.singleton(DiscoveryNodeRole.DATA_ROLE),
+            Version.V_1_0_0
+        );
+        setState(clusterService, ClusterStateCreationUtils.state(masterNode, masterNode, new DiscoveryNode[] { masterNode, dataNode }));
+
+        ClusterSettings clusterSettings = new ClusterSettings(Settings.builder().build(), ClusterSettings.BUILT_IN_CLUSTER_SETTINGS);
+        MasterTaskThrottler throttler = new MasterTaskThrottler(clusterSettings, clusterService.getMasterService(), throttlingStats);
+
+        // set some limit for update snapshot tasks
+        int newLimit = randomIntBetween(1, 10);
+
+        Settings newSettings = Settings.builder().put("master.throttling.thresholds.put-mapping.value", newLimit).build();
+
+        AtomicBoolean exceptionThrown = new AtomicBoolean();
+        try {
+            throttler.validateSetting(newSettings);
+        } catch (IllegalArgumentException e) {
+            exceptionThrown.set(true);
+        }
+        assertTrue(exceptionThrown.get());
+    }
+
+    public void testValidateSettingsForUnknownTask() {
+        DiscoveryNode masterNode = new DiscoveryNode(
+            "local_master_node",
+            buildNewFakeTransportAddress(),
+            Collections.emptyMap(),
+            Collections.singleton(DiscoveryNodeRole.CLUSTER_MANAGER_ROLE),
+            Version.V_3_0_0
+        );
+        DiscoveryNode dataNode = new DiscoveryNode(
+            "local_data_node",
+            buildNewFakeTransportAddress(),
+            Collections.emptyMap(),
+            Collections.singleton(DiscoveryNodeRole.DATA_ROLE),
+            Version.V_3_0_0
+        );
+        setState(clusterService, ClusterStateCreationUtils.state(masterNode, masterNode, new DiscoveryNode[] { masterNode, dataNode }));
+
+        ClusterSettings clusterSettings = new ClusterSettings(Settings.builder().build(), ClusterSettings.BUILT_IN_CLUSTER_SETTINGS);
+        MasterTaskThrottler throttler = new MasterTaskThrottler(clusterSettings, clusterService.getMasterService(), throttlingStats);
+
+        // set some limit for update snapshot tasks
+        int newLimit = randomIntBetween(1, 10);
+
+        Settings newSettings = Settings.builder().put("master.throttling.thresholds.random-task.value", newLimit).build();
+
+        AtomicBoolean exceptionThrown = new AtomicBoolean();
+        try {
+            throttler.validateSetting(newSettings);
+        } catch (IllegalArgumentException e) {
+            exceptionThrown.set(true);
+        }
+        assertTrue(exceptionThrown.get());
+    }
+
+    public void testUpdateThrottlingLimitForHappyCase() {
```

Comment:
Avoid happy case and use basic sanity?

Reply:
Renamed it to `testUpdateThrottlingLimitForBasicSanity`